### ISSUE TITLE
Warn on unused comparison result

### ISSUE THREAD

**Issue by WiSaGaN** [rust-lang/rust#47733]
Statements like
```
self.is_done == true;
```
are almost always typos and never intended, and it is somewhat hard to debug. We do have "bool_comparison" in clippy, but I think this error is common and significant enough that it warrants a lint directly from `rustc`. Any reason we are not doing it already?

Playground link: https://play.rust-lang.org/?gist=9122996a384f1d0fa3072915a7ccd67c&version=nightly

**Comment by rkruppe** [comment]
See #43302

**Comment by rkruppe** [comment]
Note in particular that it [warns if the feature gate is enabled](https://play.rust-lang.org/?gist=002b107defb7dddccaca9d118f720e27&version=nightly) -- arguably it should always warn, but even if that isn't fixed, the warning will come once the feature is stabilized.

**Comment by WiSaGaN** [comment]
We should be able to close this once RFC 1940 is stablised then.

**Comment by WiSaGaN** [comment]
Fixed in 1.27.0.
Result from 1.27.0 beta:
```
Compiling playground v0.0.1 (file:///playground)
warning: unused comparison which must be used
  --> src/main.rs:12:9
   |
12 |         self.foo == true;
   |         ^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_must_use)] on by default

    Finished dev [unoptimized + debuginfo] target(s) in 1.14s
     Running `target/debug/playground`
```
